### DISCUSSION TITLE
WCF Service in .NET 8 with CoreWCF ?

### DISCUSSION THREAD

**Discussion by Mike-Zarlenga-IGT** [CoreWCF/CoreWCF#1304] [Q&A] [answered]
Has anyone used CoreWCF to create a WCF Service in a .NET 8 project?

Following the instructions in Walkthrough.md in the 1.5.1 release, I'm running into several problems, the first one is, logging in the WebApplicationBuilder is indicating that port 5000 is being opened despite specifying a different port on serviceOptions.BaseAddresses.Add and serviceMetadataBehavior.HttpGetUrl:

```
app.UseServiceModel(builder =>
{
    builder.AddService<ILicenseService>((serviceOptions) => 
    {
        serviceOptions.DebugBehavior.IncludeExceptionDetailInFaults = true;
        serviceOptions.BaseAddresses.Add(new Uri("http://localhost:8889"));
    })
    .AddServiceEndpoint<ILicenseService, ILicenseService>(new BasicHttpBinding(), "/MyWCFService");
});

var serviceMetadataBehavior = app.Services.GetRequiredService<ServiceMetadataBehavior>();
serviceMetadataBehavior.HttpGetUrl = new Uri("http://localhost:8889/metadata");
serviceMetadataBehavior.HttpGetEnabled = true;
```
Log:
dbug: Microsoft.AspNetCore.Server.Kestrel[0]
      No listening endpoints were configured. Binding to http://localhost:5000 by default.
dbug: CoreWCF.Channels.ServiceModelHttpMiddleware[0]
      Building branch map
info: CoreWCF.Channels.ServiceModelHttpMiddleware[0]
      Mapping CoreWCF branch app for path /MyWCFService
dbug: CoreWCF.Channels.MetadataMiddleware[0]
      Building branch map
info: CoreWCF.Channels.MetadataMiddleware[0]
      Configuring metadata to /metadata
info: CoreWCF.Channels.MetadataMiddleware[0]
      Configuring metadata to /MyWCFService
info: Microsoft.Hosting.Lifetime[14]
      Now listening on: http://localhost:5000

Does anyone has a working example of a WCF Service in .NET 8 with CoreWCF?

**Comment by g7ed6e** [comment, marked as answer]
Hi @Mike-Zarlenga-IGT you can refer to the [ASP.NET Core documentation](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel/endpoints?view=aspnetcore-8.0) to configure the listening port.
